back-end/objects/RModelWrapper.py

```diff
@@ -41,50 +52,23 @@ def __init__(
         else:
             print("Checkpoint file not found: {}".format(net_path))
 
-    def init_model(self, network_type, pretrained, num_classes):
-        if network_type == "resnet-18":
-            self.model = torchvision.models.resnet18(
-                pretrained=pretrained, num_classes=num_classes
-            ).to(self.device)
-        elif network_type == "resnet-34":
-            self.model = torchvision.models.resnet34(
-                pretrained=pretrained, num_classes=num_classes
-            ).to(self.device)
-        elif network_type == "resnet-50":
-            self.model = torchvision.models.resnet50(
-                pretrained=pretrained, num_classes=num_classes
-            ).to(self.device)
-        elif network_type == "resnet-101":
-            self.model = torchvision.models.resnet101(
-                pretrained=pretrained, num_classes=num_classes
-            ).to(self.device)
-        elif network_type == "resnet-152":
-            self.model = torchvision.models.resnet152(
-                pretrained=pretrained, num_classes=num_classes
-            ).to(self.device)
-        elif network_type == "mobilenet-v2":
-            self.model = torchvision.models.mobilenet_v2(
-                pretrained=pretrained, num_classes=num_classes
-            ).to(self.device)
-        elif network_type == "resnet-18-32x32":
-            self.model = torchvision.models.ResNet(
-                torchvision.models.resnet.BasicBlock,
-                [2, 2, 2, 2],
-                num_classes=num_classes,
-            )
-            self.model.conv1 = torch.nn.Conv2d(
-                3, 64, kernel_size=3, stride=1, padding=1, bias=False
-            )
-            self.model.maxpool = torch.nn.MaxPool2d(kernel_size=3, stride=1, padding=1)
-            self.model = self.model.to(self.device)
-        elif network_type == "alexnet":
-            self.model = torchvision.models.alexnet(
-                pretrained=pretrained, num_classes=num_classes
-            ).to(self.device)
-        else:
-            raise NotImplementedError(
-                "Requested model type not supported. Please check."
-            )
+    def set_current_model(self, model_name: str):
+        # No change if this model is already the current model
+        if model_name == self.model_name:
+            return
+
+        # Get new model
+        new_model_data, new_model_meta_data = self.load_model_by_name(model_name)
+        if not new_model_data or not new_model_meta_data:
+            raise ValueError("Model does not exist")
+
+        # Free up current model.
+        # TODO: make sure this model is GC'ed
+        if self.model is not None:
+            del self.model
+            self.model = None
+
+        self.model_name, self.model_meta_data = new_model_data, new_model_meta_data
```

Comment:
I think here should be `self.model, self.model_name, self.model_meta_data = new_model_data, model_name, new_model_meta_data`.
Also the naming of the first variable `new_model_data` appears to be a little confusing.. Maybe `new_model`?